x/mongo/driver/operation.go

```diff
@@ -319,6 +319,14 @@ func (op Operation) selectServer(ctx context.Context) (Server, error) {
 func (op Operation) getServerAndConnection(ctx context.Context) (Server, Connection, error) {
 	server, err := op.selectServer(ctx)
 	if err != nil {
+		if op.Client != nil &&
+			!(op.Client.Committing || op.Client.Aborting) && op.Client.TransactionRunning() {
+			err = Error{
```

Comment:
Optional: Consider always wrapping all errors returned by `selectServer` in a `driver.Error`, but only adding the `TransientTransactionError` label if it's during a transaction.

Corollary question: Should all Go driver errors (eventually) satisfy the `LabeledError` interface?

Reply:
Wrapping errors from `selectServer` in `driver.Error` breaks the logic of `mongo.replaceErrors()`. I will keep the code as it is.